src/GafferImage/Display.cpp

```diff
@@ -228,23 +225,59 @@ class GafferDisplayDriver : public IECoreImage::DisplayDriver
 			return true;
 		}
 
-		ConstFloatVectorDataPtr channelData( const Imath::V2i &tileOrigin, const std::string &channelName )
+		ConstFloatVectorDataPtr channelData( const Imath::V2i &tileOrigin, const std::string &channelName, int dataCount )
 		{
 			vector<string>::const_iterator cIt = find( channelNames().begin(), channelNames().end(), channelName );
 			if( cIt == channelNames().end() )
 			{
 				return ImagePlug::blackTile();
 			}
 
-			ConstFloatVectorDataPtr tile = getTile( tileOrigin, cIt - channelNames().begin() );
-			if( tile )
+			Tile *tile = getTile( tileOrigin, cIt - channelNames().begin() );
+			if( !tile )
+			{
+				return ImagePlug::blackTile();
+			}
+
+			int previousDataCount;
+
+			{
+				tbb::spin_rw_mutex::scoped_lock tileLock( tile->mutex, false /* read */ );
+
+				previousDataCount = tile->cachedForDataCount;
+
+				if( tile->cachedForDataCount == dataCount || !tile->dirty)
+				{
+					// In order to ensure hashes and computes are consistent, once we have
+					// bound a version of the tile for this dataCount, we don't change it,
+					// even if it has been dirtied in the meantime.
+					return tile->cachedTile;
+				}
+			}
```

Comment:
I think this can be simplified a bit with `release()`, reducing vertical space and nesting, and moving the `previousDataCount` initialisation to one place, at the point we know we're actually going to need it :
```suggestion

			tbb::spin_rw_mutex::scoped_lock tileLock( tile->mutex, /* write = */ false );
                        if( tile->cachedForDataCount == dataCount || !tile->dirty)
			{
				// In order to ensure hashes and computes are consistent, once we have
				// bound a version of the tile for this dataCount, we don't change it,
				// even if it has been dirtied in the meantime.
				return tile->cachedTile;
			}
				
			int previousDataCount = tile->cachedForDataCount;
			tileLock.release();
```